src/Compiler/SetCompiler.php

```diff
@@ -0,0 +1,278 @@
+<?php
+
+/**
+ * Copyright (C) 2016 Datto, Inc.
+ *
+ * This file is part of Cinnabari.
+ *
+ * Cinnabari is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * Cinnabari is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cinnabari. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Anthony Liu <aliu@datto.com>
+ * @license http://www.gnu.org/licenses/lgpl-3.0.html LGPL-3.0
+ * @copyright 2016 Datto, Inc.
+ */
+
+namespace Datto\Cinnabari\Compiler;
+
+use Datto\Cinnabari\Exception\ArgumentsException;
+use Datto\Cinnabari\Exception\CompilerException;
+use Datto\Cinnabari\Format\Arguments;
+use Datto\Cinnabari\Mysql\Expression\Column;
+use Datto\Cinnabari\Mysql\Expression\Parameter;
+use Datto\Cinnabari\Mysql\Update;
+use Datto\Cinnabari\Translator;
+
+/**
+ * Class SetCompiler
+ * @package Datto\Cinnabari
+ */
+class SetCompiler extends AbstractCompiler
+{
+    /** @var Update */
+    protected $mysql;
+    
+    public function compile($translatedRequest, $arguments)
+    {
+        $this->request = $translatedRequest;
+
+        $this->mysql = new Update();
+        $this->arguments = new Arguments($arguments);
+
+        if (!$this->enterTable($hasZero)) {
```

Comment:
We should remove this $hasZero variable, here and elsewhere, since it is no longer necessary.